clients/src/main/java/org/apache/kafka/clients/producer/ProducerRecord.java

```diff
@@ -57,7 +57,11 @@ public ProducerRecord(String topic, Integer partition, Long timestamp, K key, V
         if (topic == null)
             throw new IllegalArgumentException("Topic cannot be null");
         if (timestamp != null && timestamp < 0)
-            throw new IllegalArgumentException("Invalid timestamp " + timestamp);
+            throw new IllegalArgumentException(
+                    String.format("Invalid timestamp: %d. Timestamp should always be non-negative or null ", timestamp));
```

Comment:
Nit: there's a space at the end of the String. We should probably be consistent and have a period at the end of all exception messages in this constructor.